### DISCUSSION TITLE
Error when increasing $MXSTACK in array_sizes.h file in egs_chamber

### DISCUSSION THREAD

**Discussion by Fisicomelo** [nrc-cnrc/EGSnrc#1190] [Questions] [answered]
Hello guys, 

I am trying to run egs_chamber with a BEAM accel as shared library, and I am constantly getting this error:
Stack overflow in BREMS! np = ****** Increase $MXSTACK and try again
However, if I try to increase $MXSTACK from 9e6 to 1e7 in the array_sizes.h file in egs_chamber and try to recompile I get another error (please, see image) that does not allow me to compile it:
![compiling_error](https://github.com/user-attachments/assets/09ddf74d-8609-48e8-b8d2-20c09b2b290e)

I have tried to increase $MXSTACK in /home/omega/EGSnrc/HEN_HOUSE/src/egsnrc.macros as well but I still have the same (first) error.

Does anyone have any suggestion?

DBS settings:
Splitting field radius: 10
SSD: 100
Beam splitting number: 800

**Comment by rtownson** [comment]
I haven't been able to reproduce this. Does your `array_sizes.h` file look like this?
```
#ifndef ARRAY_SIZES_
#define ARRAY_SIZES_

#define MXMED   40
#define MXSTACK 1000000

#endif
```
And egsnrc.macros:
```
PARAMETER $MXSTACK=1000000;    "STACK SIZE"
```

Have you made any other changes?

**Comment by Fisicomelo** [comment]
Hi Reid,
Yes, it is except for the value of $MXSTACK. In your case it is 1e6 and I am trying to increase up to 1e7 and that is when I can't compile anymore. I can compile egs_chamber only if I increase up to 9999999, but then when I try to run the code, I get the error massage again saying that I need to increase $MXSTACK. 
I believe the version installed at te University cluster is from 2021. I will install my own version in home area and try again.
Thank you, Reid

**Reply by rtownson** [reply]
Thanks for reporting this! This was a simple bug, the fix is here: 

https://github.com/nrc-cnrc/EGSnrc/pull/1193

**Reply by Fisicomelo** [reply]
I got rid of the "compiling problem" by doing the modification you suggested. However, increasing $MXSTACK to 1e7 was still not enough (same initial error: Stack overflow in BREMS! np = ****** Increase $MXSTACK and try again) 
If I increase $MSTACK to 1e9 and run egs_chamber again, it solves the first error but it gives another problem related to allocatin memory (image attached).
<img width="784" alt="memory_error" src="https://github.com/user-attachments/assets/0e7b4ff4-ea4c-448e-9115-ea369a66168d">

**Reply by rtownson** [reply]
Try reducing the splitting number, you probably still will get decent efficiency with a lower number like 100 instead of 800. This will reduce the MXSTACK required. Then find the minimum MXSTACK that doesn't give you the memory error. Otherwise, you will need to install more RAM in the computer (or find a different computer).

**Reply by Fisicomelo** [reply]
Hi Reid,

Sure, I will try that. Thank you so much for the support.

**Reply by Fisicomelo** [reply]
Hi @rtownson ,

I tried to run a very small number of histories in egs_chamber (1e4) and without splitting in my accelerator (DBS = none) I still get the same error asking to increase $MXSTACK. I guess the problem is probably in the accelerator that I built, although I have already tried to re-build many times.
Let me know if you have any clue.

Thank you.

**Reply by rtownson** [reply]
Do you have any issues running the BEAMnrc accelerator on its own? Can you run the egs_chamber simulation with a different BEAMnrc accelerator? E.g. using BEAM_EX16MVp.

**Reply by Fisicomelo** [reply]
Hi Reid,
 I forgot to mention. That is the weird thing. There is no issue when I run the accelerator (this Elekta one that I built) itself. It runs just fine. It also works to run egs_chamber   with other accelerators such as the Co60 unit that I also built it. I guess next steps would be to run the same egs_chamber inputfile with the BEAM_EX16MVp as you suggested and also run a different egs_chamber inputfile (some of the examples) with the Elekta accel. Thank you for the suggestions. I will keep you upadted.

**Reply by Fisicomelo** [reply, marked as answer]
Hello @rtownson 

I have performed some tests, and the source of problem was probably my egs_chamber input file that for some reason was corrupted. I created a new inputfile from another one that was fine and now it works. It is still weird why it flagged an error message to increase $MXSTACK, but at least everything is fine now.
Thank you so much for the support.

Cheers

**Comment by Fisicomelo** [comment]
Fantastic! Thank you, Reid. I really appreciate it.
